### ISSUE TITLE
When ever I run "git clone https://github.com/spences10/twitter-bot-bootstrap" I seem to be missing the folder "helpers" and the files within

### ISSUE THREAD

**Issue by DaveBryantLU** [spences10/twitter-bot-bootstrap#68]
As the title suggests, I run ""git clone https://github.com/spences10/twitter-bot-bootstrap"" and all seems to work ok, however the folder "helpers" with files <sentiment.js> & <strings.js> are missing. 

This seems to hinder my install as even after following each step after and entering "npm run" nothing is retweeted on my feed. 

This is the message I receive after an "NPM RUN" 

davebryantlu:~/workspace/twitter-bot-bootstrap (master) $ npm run
Lifecycle scripts included in twitter-bot-bootstrap:
  start
    node index.js
  test
    snyk test && standard && node index.js
  prepublish
    npm run snyk-protect

available via `npm run-script`:
  snyk-protect
    snyk protect
  deploy
    now -E
davebryantlu:~/workspace/twitter-bot-bootstrap (master) $ 

Any idea where I have gone wrong ?

**Comment by spences10** [comment]
Hi @DaveBryantLU thanks for logging this, can I ask are you following the guide from the Medium post?

I don't think there's an instruction to use just `npm run` it would be either `npm start` or `npm run start` so if you can clarify that then I'll be able to fix that for everyone else.

If you're coming from the medium post, that post and this repo are a bit different now. Due to the Twitter ToS changing #48 I have had to change the bot functionality with #51 

**Tl;Dr**: this means that there's no like functionality so no need for the sentiment detection and the strings are now configured through the `.env` file

Take a look at the [README.md](https://github.com/spences10/twitter-bot-bootstrap#twitter-bot-bootstrap) on this repo for a walk through on how I'm doing it now.

If you find any more issues please feel free to log another issue.

Thanks again

**Comment by DaveBryantLU** [comment]
Thanks for the response! I should clarify I was using "NPM START" and also tried "NPM RUN START", that was a typo on my last. I'll go through the new README file above and let you know if I run into any further issues, thanks again for such a comprehensive guide.

**Comment by DaveBryantLU** [comment]
Hi there, I restarted the setup from scratch and now receive a difference error, the error is "Failed at the twitter-bot-bootstrap@0.2.1 start script 'node index.js'." I've attached the log, I've rechecked your latest README file several times and followed it to the letter (API/Consumer Keys and Search strings aside) 

Any further insight here would be appreciated
[npm-debug.pdf](https://github.com/spences10/twitter-bot-bootstrap/files/1091244/npm-debug.pdf)

**Comment by spences10** [comment]
HI @DaveBryantLU so you're using C9 to set up the bot yes?

It looks like you're using node v4 form the log, let me go see if I can spin up a new bot, bear with!

**Comment by spences10** [comment]
Ok, so, first off it looks like using a more recent version of node will stop that error but it looks like something that has changed in the most recent version of `twit` is causing errors now as it's appears to be silently failing on the retweet search.

So in answer to the issues posted earlier use `nvm` to go to a more recent version of node, I went to v8, with this:

```shell
nvm install 8
nvm use 8
```

This is assuming you're using c9 as `nvm` is installed by default.

With regard to the other error you will get after that `ERRORDERP: Cannot assign retweeID`, that needs to be fixed, I'm not sure what it is and I have no time to look into it right now.

If you work out what it is please feel free to open a Pull Request @DaveBryantLU 

If you need a bit more granular information take a look at my [Twitter bot playground][tbp] but I can't guarantee that you will not have the same issues as with the bootstrap here 🙃 

[tbp]: https://github.com/spences10/twitter-bot-playground#twitter-bot-playground

**Comment by DaveBryantLU** [comment]
First error fixed, I'm using c9 as I find that's the easiest method of setting up stuff like this quickly. 

If I happen to work out the `ERRORDERP: Cannot assign retweeID` issue I'll let you know @spences10 ! Have a great day!

**Comment by spences10** [comment]
Hey @DaveBryantLU I found what was causing the no search results it was the `LANG` setting, I haven't found the precise cause but basically if you had `en` for your language configured it was returning the `process.env.lang` which is a `node` setting, I have changed the config [here](47835da6cf151b55395d8083faa3f815c2e65aa3) and the correcspondin `.env` setting will need changing too.

Let me know if that fixes it for you

thanks

**Comment by DaveBryantLU** [comment]
Hi Scott,

I restarted, pulled down the latest version, updated the .env file to show
"LANGUAGE=en" and not "LANG" but still getting the "Cannot assign retweet
ID" message at the last stage.

I'll have a scan of this tonight and see what I can come up with, quite
strange. I'm fairly new to this, but do have a basic understanding and have
followed your instructions to the letter.

Appreciated your help so far!

Best
Dave


[image: David-Bryant-174x174.jpeg]

David Bryant, David@learnupon.com

Account Manager Australia, LearnUpon

Phone +61 450 745 280

@learnupon <https://twitter.com/learnupon> | http://www.learnupon.com/
<http://www.learnupon.com/>

Find out why we have the best LMS in Australia here.
<https://www.learnupon.com/learnupon-best-lms-australia/>


<https://www.learnupon.com/webinar-4-ingredients-great-lms/?utm_source=email_signature&utm_medium=email&utm_campaign=signature_ingredients_webinar&utm_term=ingredients_webinar>

**Comment by spences10** [comment]
The only other place to change it is in the `config.js` file, that's where the environment variables are accessed. 

If you are using c9 and are on a public machine you can DM me the link on twitter or email it over to me and I'll take a look:

[@ScottDevTweets](https://twitter.com/ScottDevTweets)
[spences10apps@gmail.com](mailto:spences10apps@gmail.com)
